### PR TITLE
Add rolling update max unavailable to CNI chart to speed up deploys (cherry pick to release-1.18)

### LINKED ISSUE #44859: [release-1.18] Add rolling update max unavailable to CNI chart to speed up deploys
Manual cherrypick required.

#44839 failed to apply on top of branch "release-1.18":
```
Applying: Add rolling update max unavailable to CNI chart to speed up deploys
Using index info to reconstruct a base tree...
M	manifests/charts/istio-cni/values.yaml
M	operator/pkg/apis/istio/v1alpha1/values_types.proto
Falling back to patching base and 3-way merge...
Auto-merging operator/pkg/apis/istio/v1alpha1/values_types.proto
CONFLICT (content): Merge conflict in operator/pkg/apis/istio/v1alpha1/values_types.proto
Auto-merging manifests/charts/istio-cni/values.yaml
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch=diff' to see the failed patch
Patch failed at 0001 Add rolling update max unavailable to CNI chart to speed up deploys
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".

```

### PR DESCRIPTION
**Pull request by myshkin5**
**Please provide a description of this PR:**
Updates to the CNI chart take a long time because the default is to update one node at a time. This PR allows the max unavailable to be overridden so multiple node can be updated at once.

Other charts expose a similar override. Note that I did not expose max surge as it seemed inappropriate to have more than one CNI pod per node even if just during an update.

Fixes #44859

### CODE REVIEW

File: operator/pkg/apis/istio/v1alpha1/values_types.proto [hunk 1]
```diff
@@ -88,6 +88,11 @@ message CNIConfig {
   google.protobuf.Struct seccompProfile = 19;
 
   CNIAmbientConfig ambient = 21;
+
+  string provider = 22;
```
Comment (hanxiaop) [inline operator/pkg/apis/istio/v1alpha1/values_types.proto:92]: It looks like this is extra change which is not included in the original PR https://github.com/istio/istio/pull/44839

Comment (hanxiaop) [reply operator/pkg/apis/istio/v1alpha1/values_types.proto:92]: Can you verify if anything should be cherry-picked before this PR?

Comment (hanxiaop) [reply operator/pkg/apis/istio/v1alpha1/values_types.proto:92]: I see, there's another PR that should be cherry-picked before yours. I'm going to put on a hold before https://github.com/istio/istio/issues/44642 is resolved.

Comment (myshkin5) [reply operator/pkg/apis/istio/v1alpha1/values_types.proto:92]: @hanxiaop Thanks for catching this. I'll rebase this PR when #44642 is resolved.

Comment (myshkin5) [reply operator/pkg/apis/istio/v1alpha1/values_types.proto]: @hanxiaop Since it looks like #44642 may not be included in 1.18, I just reserved the field number in the proto

**Review by jacob-delgado — APPROVED** [review]
I like the use of reserved for provider = 22.

File: manifests/charts/UPDATING-CHARTS.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ These manifests can be found in [gen-istio.yaml](../charts/istio-control/istio-d
 To regenerate the manifests, run:
 
 ```bash
-$ make copy-templates gen-kustomize update-golden
```
Comment (jacob-delgado) [inline manifests/charts/UPDATING-CHARTS.md:62]: why was this removed?

Comment (hanxiaop) [reply manifests/charts/UPDATING-CHARTS.md:62]: Yes I think encountered the same thing before

**Review by hanxiaop — APPROVED** [review]

**Comment by hanxiaop** [comment]
> I like the use of reserved for provider = 22.

I think this has resolved my issue.
